From 1047121c46e34749a92bf7e7f5193d4fcbe65e52 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Thu, 28 Feb 2013 12:52:12 +0000
Subject: [PATCH] Disable virnettlscontexttest.

---
 tests/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index d3a7868..1226a41 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -106,7 +106,7 @@ test_programs = virshtest sockettest \
 	$(NULL)
 
 if WITH_GNUTLS
-test_programs += virnettlscontexttest
+#test_programs += virnettlscontexttest
 endif
 
 if WITH_SECDRIVER_SELINUX
-- 
1.8.1.2

