diff -rup libvirt-3.4.0.orig/gnulib/lib/getopt-core.h libvirt-3.4.0.new/gnulib/lib/getopt-core.h
--- libvirt-3.4.0.orig/gnulib/lib/getopt-core.h	2017-05-27 17:19:29.311435569 +0100
+++ libvirt-3.4.0.new/gnulib/lib/getopt-core.h	2017-06-05 11:07:41.501442780 +0100
@@ -17,8 +17,8 @@
    License along with the GNU C Library; if not, see
    <http://www.gnu.org/licenses/>.  */
 
-#ifndef _GETOPT_CORE_H
-#define _GETOPT_CORE_H 1
+#ifndef _GL_GETOPT_CORE_H
+#define _GL_GETOPT_CORE_H 1
 
 /* This header should not be used directly; include getopt.h or
    unistd.h instead.  Unlike most bits headers, it does not have
@@ -93,4 +93,4 @@ extern int getopt (int ___argc, char *co
 
 __END_DECLS
 
-#endif /* _GETOPT_CORE_H */
+#endif /* _GL_GETOPT_CORE_H */
diff -rup libvirt-3.4.0.orig/gnulib/lib/getopt-ext.h libvirt-3.4.0.new/gnulib/lib/getopt-ext.h
--- libvirt-3.4.0.orig/gnulib/lib/getopt-ext.h	2017-05-27 17:19:29.316435518 +0100
+++ libvirt-3.4.0.new/gnulib/lib/getopt-ext.h	2017-06-05 11:07:51.806408528 +0100
@@ -17,8 +17,8 @@
    License along with the GNU C Library; if not, see
    <http://www.gnu.org/licenses/>.  */
 
-#ifndef _GETOPT_EXT_H
-#define _GETOPT_EXT_H 1
+#ifndef _GL_GETOPT_EXT_H
+#define _GL_GETOPT_EXT_H 1
 
 /* This header should not be used directly; include getopt.h instead.
    Unlike most bits headers, it does not have a protective #error,
@@ -74,4 +74,4 @@ extern int getopt_long_only (int ___argc
 
 __END_DECLS
 
-#endif /* _GETOPT_EXT_H */
+#endif /* _GL_GETOPT_EXT_H */
